### DISCUSSION TITLE
TanStack Start: How to access environment variables when process.env is not present?

### DISCUSSION THREAD

**Discussion by marcfalk** [TanStack/router#2591] [Q&A]
TanStack Start looks super interesting 🙌

Coming from [Remix](https://remix.run/), for which hosting platform adapters are pretty mature at this point, there are plenty of examples on how to get your site up and running on any one platform. For example on Cloudflare Pages with Cloudflare D1 for SQLite database and Drizzle (ORM) in the middle. I'm wondering how one would go about that with TanStack Start.

From the docs I can't find much about accessing environment variables - e.g. database connection strings - unless `process.env` is present. I found some examples in the docs under _Start Examples_ (e.g. [this one](https://tanstack.com/router/latest/docs/framework/react/examples/start-supabase-basic)). When `process.env` is not present what should I do to access variables provided by the hosting platform?

**Comment by neoantox** [comment]
Hi, @marcfalk! Have you found a way to access cloudflare env variables? I'm also trying to understand how to access them

**Comment by dan-gamble** [comment]
[Here](https://github.com/timoconnellaus/tanstack-start-workers) is a working Cloudflare version

**Comment by germancasares** [comment]
You can use `import.meta.env.VITE_API_BASE_URL` they key here being prepending the env var with VITE_ so that the env var is exposed to the client.

[To prevent accidentally leaking env variables to the client, only variables prefixed with VITE_ are exposed to your Vite-processed code.](https://vite.dev/guide/env-and-mode#env-variables)
